mesh/mesh.hpp

```diff
@@ -880,42 +985,27 @@ class Mesh
       // Finalize(); // reminder: not needed
    }
 
-   /** Creates mesh by reading a file in MFEM, Netgen, or VTK format. If
-       generate_edges = 0 (default) edges are not generated, if 1 edges are
-       generated. See also @a Mesh::LoadFromFile. */
-   explicit Mesh(const char *filename, int generate_edges = 0, int refine = 1,
-                 bool fix_orientation = true);
-
-   /** Creates mesh by reading data stream in MFEM, Netgen, or VTK format. If
-       generate_edges = 0 (default) edges are not generated, if 1 edges are
-       generated. */
-   explicit Mesh(std::istream &input, int generate_edges = 0, int refine = 1,
-                 bool fix_orientation = true);
-
-   /// Create a disjoint mesh from the given mesh array
-   ///
-   /// @note Data is copied from the meshes in @a mesh_array.
-   Mesh(Mesh *mesh_array[], int num_pieces);
-
    /// Deprecated: see @a MakeRefined.
    MFEM_DEPRECATED
    Mesh(Mesh *orig_mesh, int ref_factor, int ref_type);
 
-   /** This is similar to the mesh constructor with the same arguments, but here
-       the current mesh is destroyed and another one created based on the data
-       stream again given in MFEM, Netgen, or VTK format. If generate_edges = 0
-       (default) edges are not generated, if 1 edges are generated. */
-   /// \see mfem::ifgzstream() for on-the-fly decompression of compressed ascii
-   /// inputs.
-   virtual void Load(std::istream &input, int generate_edges = 0,
-                     int refine = 1, bool fix_orientation = true)
-   {
-      Loader(input, generate_edges);
-      Finalize(refine, fix_orientation);
-   }
+   /// @}
 
-   /// Clear the contents of the Mesh.
-   void Clear() { Destroy(); SetEmpty(); }
+   /// @name Information about the mesh as a whole
+   /// @{
+
+   /// @brief Dimension of the reference space used within the elements
+   int Dimension() const { return Dim; }
+
+   /// @brief Dimension of the physical space containing the mesh
```

Comment:
May want to mention that this can be bigger than the dimension for eg. surface meshes in 3D.